docs/proposals/11-cluster-health.md

```diff
@@ -0,0 +1,205 @@
+The list of affected zones could be maintained in the [`Message`](https://github.com/kubernetes/cluster-registry/blob/master/pkg/apis/clusterregistry/v1alpha1/types.go#L168).
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty if this condition lasts for some duration.
+- If necessary, `kube-controller-manager` could be scaled down as described [above](#possible-actions).
+- Some resource creations (say, `pods`) could be throttled/rate-limited.
+- The `machine-controller-manager` could be either scaled down as described [above](#possible-actions) or be enhanced to avoid (or ramp down) deletion of machines from the affected zones (or `machinedeployments`).
+- Gardener can defer any disruptive updates until later.
+
+### Condition Types of future relevance
+
+This is just a sample set of condition types that might be of future relevance. It is not an exhaustive list.
+The suggested possible actions for these condition types are also just samples and not an exhaustive list.
+
+#### UnhealthyInternalNetwork
+
+If there is a way to detect problems in the cluster network that is severe enough to affect any significant portion of the cluster then this could be encoded in a component that updates the conditions with the `UnhealthyInternalNetwork` condition type.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
+- Some resource creations (say, `pods`) could be throttled/rate-limited.
+- Gardener can defer any disruptive updates until later.
+
+#### UnhealthyCloudProvider
+
+If there is information about some issues reported against the cloud provider, it could be updated in the conditions with the `UnhealthyCloudProvider` condition type.
+If there is more specific information about the issue then more specific condition types can be defined.
+
+Such conditions could be updated manually by the administrators, operators on duty based on publications of such issues by the cloud providers.
+If there is a way to automatically scrape such information from well-known locations or APIs then this could even be automated.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
+- Controllers such as `kube-controller-manager`, `machine-controller-manager` that manage crucial resources such as `nodes` can be either scaled down as describe [above](#possible-actions) or enhanced to change their behaviour to ramp down their active interventions to the cluster.
+- Some resource creations (say, `pods`) could be throttled/rate-limited.
+- Gardener can defer any disruptive updates until later.
+
+#### CloudProviderQuotaExceeded
+
+If there is a way to detect that some cloud provider quotas are exceeded this information can be updated in the conditions with the `CloudProviderQuotaExceeded` condition type.
```

Comment:
I found this condition very useful.
Could you propose implementation design?
The message can contain more detailed information about what quotas are reached, e.g. VM counts, CPU, Memory, LoadBalancers, Disks and so on.

Reply:
> Could you propose implementation design?

I was more going for an agreement about the approach rather than individual conditions and implementations. But sure, we can pick this as part of the first set of conditions and actions to be implemented.